src/installation/index.md

```diff
@@ -1,61 +1,170 @@
 
-Let's take a moment to discuss the Rust support for the architectures of the Espressif chips in more detail. At this moment, Espressif SoCs are based on two different architectures: `RISC-V` and `Xtensa`. The support for those two architectures in the Rust programming language is very different.
+Let's take a moment to discuss the Rust support for the different architectures of the Espressif chips and the different approaches in more detail. At this moment, Espressif SoCs are based on two different architectures: `RISC-V` and `Xtensa`. The support for those two architectures in the Rust programming language is very different.
 
-## Rust in RISC-V targets
+[Ecosystem Overview]: ../overview/index.md
 
-The `RISC-V` architecture has support in the mainline Rust compiler so setup is relatively simple, all we must do is add the appropriate compilation target.
+## RISC-V targets
 
-Hence, the dependencies required to develop Rust applications in `RISC-V` targets are:
-- Rust Toolchain with the proper target: Used to compile our Rust code.
-- `LLVM`: Used as codegen backend by the Rust compiler.
-- \[_Optional_\] `GCC` Toolchain: `GCC` linker can be used.
-  - `GCC` is marked as optional as you can also use `LLVM` linker.
+The `RISC-V` architecture has support in the mainline Rust compiler so, the setup is relatively simple. There are two ways of proceeding with the installation:
+- Using [`espup`, a tool that will be covered later]
+- Using the official Rust tools
 
-Aditonally when building `std` applications we also need:
-- [ESP-IDF]: Espressif IoT Development Framework.
-- [`ldproxy`] crate:  Simple tool to forward linker arguments given to [`ldproxy`] to the actual linker executable. The crate can be found in the [esp-rs/embuild] repository.
+If you only want to use `RISC-V` targets, you can use the official Rust tools, for this approach we need [`rustup`] installed, and a [Rust nightly toolchain] with the `rust-src` [component]. We can install a nightly toolchain with the `rust-src` component via:
+
+```bash
+rustup toolchain install nightly --component rust-src
+```
+
+This are the two recommended targets for most Espressif `RISC-V` chips:
+
+- For bare-metal (`no_std`) applications: `riscv32imc-unknown-none-elf`
+- For applications that require `std`: `riscv32imc-esp-espidf`
+
+> #### A note in RISC-V `no_std` Rust targets.
+>
+> There are 3 different flavors of RISC-V 32 target in Rust:
+> - `riscv32i-unknown-none-elf`
+> - `riscv32imc-unknown-none-elf`
+> - `riscv32imac-unknown-none-elf`
+>
+> Where:
+> - `32i`: Base Integer Instruction Set for 32 bits
+> - `m`: Standard Extension for Integer Multiplication and Division
+> - `a`: Standard Extension for Atomic Instructions
+>   - Only available in ESP32-S2 at the moment
```

Comment:
guess ESP32-S2 doesn't belong here

Reply:
Upps, atomics are supported on ESP32 and ESP32-S3, right?

Good catch!

Reply:
Yeah, but since this is the chapter about RISCV no need to mention them here
C6 supports the A extension AFAIK